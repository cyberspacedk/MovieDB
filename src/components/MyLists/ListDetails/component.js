import React from 'react';
import PropTypes from 'prop-types';
import PageLayout from '../../shared/Layout';

const ListDetails = ({
  loading,
  empty,
  error,
  listDetails,
  history,
  listName,
}) => {
  return (
    <PageLayout
      title={listName}
      loading={loading}
      empty={empty}
      error={error}
      array={listDetails}
      history={history}
    />
  );
};

ListDetails.propTypes = {
  listDetails: PropTypes.arrayOf(PropTypes.object).isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.bool.isRequired,
  history: PropTypes.isRequired,
  empty: PropTypes.bool.isRequired,
  listName: PropTypes.string.isRequired,
};

export default ListDetails;
