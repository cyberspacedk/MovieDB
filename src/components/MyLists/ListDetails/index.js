import ListDetailsContainer from './container';

export default ListDetailsContainer;
