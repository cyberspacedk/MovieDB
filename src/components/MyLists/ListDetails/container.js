/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ListDetails from './component';
import { getListDetailsRequest } from '../../../store/listDetails/actions';
import {
  getListDetails,
  isError,
  isLoading,
  isEmpty,
} from '../../../store/listDetails/selectors';

class ListDetailsContainer extends Component {
  componentDidMount() {
    const { id } = this.props.match.params;
    this.props.getListDetailsRequest(id);
  }

  render() {
    return <ListDetails {...this.props} />;
  }
}

ListDetailsContainer.propTypes = {
  id: PropTypes.number.isRequired,
  getListDetailsRequest: PropTypes.func.isRequired,
  match: PropTypes.shape(PropTypes.object).isRequired,
};

const mstp = state => ({
  loading: isLoading(state),
  error: isError(state),
  empty: isEmpty(state),
  listDetails: getListDetails(state),
});

const mdtp = {
  getListDetailsRequest,
};
export default connect(
  mstp,
  mdtp,
)(ListDetailsContainer);
