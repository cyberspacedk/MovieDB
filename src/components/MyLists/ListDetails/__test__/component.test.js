import React from 'react';
import { shallow } from 'enzyme';
import ListDetails from '../component';

describe('Component: ListDetails', () => {
  const props = {
    loading: false,
    empty: false,
    error: false,
    detailsList: [{ id: 1 }],
    history: {
      push: jest.fn(),
    },
    listName: 'Comedy',
    deleteMovieFromListRequest: jest.fn(),
  };

  const container = shallow(<ListDetails {...props} />);

  it('Snapshot: should match', () => {
    expect(container).toMatchSnapshot();
  });
});
