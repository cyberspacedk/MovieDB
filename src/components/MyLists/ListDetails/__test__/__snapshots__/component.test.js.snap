// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ListDetails Snapshot: should match 1`] = `
<PageLayout
  array={
    Array [
      Object {
        "id": 1,
      },
    ]
  }
  empty={false}
  error={false}
  history={
    Object {
      "push": [MockFunction],
    }
  }
  loading={false}
  operations={[MockFunction]}
  removeBox={true}
  title="Comedy"
/>
`;
